lib/hammer_cli/testing/output_matchers.rb

```diff
@@ -0,0 +1,113 @@
+module HammerCLI
+  module Testing
+    module OutputMatchers
+      class MatcherBase
+        include MiniTest::Assertions
+
+        def initialize(expected="")
+          @expected = expected
+        end
+
+        def assert_match(actual)
+          assert_equal(@expected, actual)
+        end
+      end
+
+      class FieldMatcher < MatcherBase
+        def initialize(label, value)
+          @expected = FieldMatcher.matcher(label, value)
+        end
+
+        def assert_match(actual)
+          message = "Regex /#{@expected.source}/ didn't match the output:\n#{actual}"
+          assert(@expected =~ actual, message)
+        end
+
+        def self.matcher(label, value)
+          Regexp.new(Regexp.quote(label) + ':[ ]+' + Regexp.quote(value))
+        end
+      end
+
+      class OutputMatcher < MatcherBase
+        attr_accessor :expected_lines
+
+        def initialize(expected="", options={})
+          @expected_lines = expected.is_a?(Array) ? expected : [expected]
+          @ignore_whitespace = options[:ignore_whitespace].nil? ? true : options[:ignore_whitespace]
+        end
+
+        def assert_match(actual)
+          if @ignore_whitespace
+            expected_lines = striped_lines
+            actual = strip_output(actual)
+          else
+            expected_lines = @expected_lines
+          end
+          expected_lines = expected_lines.join("\n")
+
+          message = "Output didn't contain expected lines:\n" + diff(expected_lines, actual)
+          assert(actual.include?(expected_lines), message)
+        end
+
+        protected
+
+        def striped_lines
+          @expected_lines.map do |line|
+            line.gsub(/\s*$/, '')
+          end
+        end
```

Comment:
Could be more DRY and I assume rstrip is more efficient:
```ruby
def strip_lines(lines)
  lines.map(&:rstrip)
end
```
and reuse in `strip_lines(@expected_lines)` and `strip_lines(actual_split("\n")).join("\n")`